### PR TITLE
boards: Add BL653DVK support

### PR DESCRIPTION
**Pull request by greg-leach**
Add support for the Laird Connectivity BL653 DVK

Signed-off-by: Greg Leach <greg.leach@lairdconnect.com>

### CODE REVIEW

File: boards/arm/bl653_dvk/doc/bl653_dvk.rst [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+***********
+
+The following approach can be used when an application needs to use
+more than one UART for connecting peripheral devices:
+
+1. Add devicetree overlay file to the main directory of your application:
+
+   .. code-block:: console
+
+      $ cat nrf52833dk_nrf52833.overlay
```
Comment (rerickson1) [inline boards/arm/bl653_dvk/doc/bl653_dvk.rst:184]: should be `bl653_dvk.overlay`

File: boards/arm/bl653_dvk/doc/bl653_dvk.rst [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+   In the overlay file above, pin P0.16 is used for RX and P0.14 is used for TX
+
+2. Use the UART1 as ``device_get_binding(DT_LABEL(DT_NODELABEL(uart1)))``
+
+Overlay file naming
+===================
+
+The file has to be named ``<board>.overlay`` and placed in the app main directory to be
+picked up automatically by the build system.
```
Comment (rerickson1) [inline boards/arm/bl653_dvk/doc/bl653_dvk.rst:201]: Maybe link to this as well: https://docs.zephyrproject.org/latest/guides/dts/howtos.html#set-devicetree-overlays
It describes the options you have for specifying overlays and where they can be located.

**Review by rerickson1 — CHANGES_REQUESTED** [review]

**Review by ioannisg — APPROVED** [review]
The board definition seems legitimate to me.

File: boards/arm/bl653_dvk/doc/bl653_dvk.rst (hunk 1 above)
Comment (greg-leach) [reply boards/arm/bl653_dvk/doc/bl653_dvk.rst]: Updated.

File: boards/arm/bl653_dvk/doc/bl653_dvk.rst (hunk 2 above)
Comment (greg-leach) [reply boards/arm/bl653_dvk/doc/bl653_dvk.rst:201]: Updated.

**Review by rerickson1 — APPROVED** [review]

**Pull request merged**